File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
export class Squares {
    #count;

    /** @param {number} count */
    constructor(count) {
        this.#count = count;
    }

    get sumOfSquares() {
        let sum = 0;
        for (let i = 1; i <= this.#count; i++) {
            sum += i * i;
        }
        return sum;
    }

    get squareOfSum() {
        let sum = 0;
        for (let i = 1; i <= this.#count; i++) {
            sum += i;
        }
        return sum * sum;
    }

    get difference() {
        return this.squareOfSum - this.sumOfSquares;
    }
}

File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
class Squares
  def initialize(number)
    @number = number
  end

  def square_of_sum
    (@number * (@number + 1) / 2) ** 2
  end

  def sum_of_squares
    (@number * (@number + 1) * (2 * @number + 1)) / 6
  end

  def difference
    square_of_sum - sum_of_squares
  end
end